lua/dap.lua

```diff
@@ -347,6 +347,27 @@ function Session:_request_scopes(current_frame)
   end)
 end
 
+function Session:_goto(line, source, col)
+  local frame = self.current_frame
+  if frame then
+    self:request('gotoTargets',  {source = source or frame.source, line = line, col = col}, function(err, response)
+      if err then
+        print(err.message)
+      end
+      if response then
+        if response.targets then
+          self:request('goto', {threadId = session.stopped_thread_id, targetId = response.targets[1].id },
+          function(err, _)
+            if err then
+              print(err.message)
+            end
+          end)
+        end
+      end
+    end)
+  end
```

Comment:
I think it would help with the readability to add some early returns, to reduce the number of indentation levels a bit.

And it might make sense to check for `supportsGotoTargetsRequest` and give the user a message if it's not supported.

```suggestion
  local frame = self.current_frame
  if not frame then
    return
  end
  if not self.capabilities.supportsGotoTargetsRequest then
    print('Debug Adapter doesn't support GotoTargetRequest')
    return
  end
  self:request('gotoTargets',  {source = source or frame.source, line = line, col = col}, function(err, response)
    if err then
      print('Error getting gotoTargets: ' .. err.message)
      return
    end
    if not response or not response.targets then
      print('No goto targets available. Can't execute goto')
      return
    end
    local params = {threadId = session.stopped_thread_id, targetId = response.targets[1].id }
    self:request('goto', params, function(err, _)
      if err then
        print('Error executing goto: ' .. err.message)
      end
    end)
  end)

```